takanon_extractions/dynamic_extractions/extract_content.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+"""
+Extract content for specific section types from HTML files based on structure JSON.
+"""
+
+import argparse
+import json
+import sys
+from pathlib import Path
+from bs4 import BeautifulSoup
+from markdownify import markdownify as md
+from botnim.config import get_logger
+
+# Logger setup
+logger = get_logger(__name__)
+
+def extract_content_for_sections(html_content, structure_data, target_content_type):
+    """
+    Extract complete content for sections of the specified type.
+    
+    Args:
+        html_content: Raw HTML content
+        structure_data: Parsed JSON structure
+        target_content_type: Type of content to extract (e.g., "סעיף")
+    
+    Returns:
+        Updated structure with content added
+    """
+    soup = BeautifulSoup(html_content, 'html.parser')
+    
+    # Find all sections with html_id in the structure
+    sections_with_ids = []
+    
+    def collect_sections(items):
+        for item in items:
+            if item.get('html_id'):
+                sections_with_ids.append(item)
+            if 'children' in item:
+                collect_sections(item['children'])
+    
+    collect_sections(structure_data['structure'])
+    
+    # Sort sections by their position in the HTML
```

Comment:
This is very ineffective. 
A better way would have been to find all html elements with `id`s, store the ids in a list and then sort `sections_with_positions` with a customized `key` function:

```python
 all_ids = <get it from soup>
 sections_with_ids.sort(key=lambda x: all_ids.index(x['html_id'])
 ```